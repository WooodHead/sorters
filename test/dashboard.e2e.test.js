import cheerio from 'cheerio'
import pretty from 'pretty'
import {ObjectID} from 'mongodb'
import {setup, teardown} from './setup'
import setCookie from 'set-cookie-parser'
import {generateAndLogUser} from './fixtures'

let db
let browser

beforeAll(async () => {
    const res = await setup()
    db = res.db
    browser = res.browser
})

afterEach(async () => {
    await db.dropDatabase()    
})

afterAll(async () => {
    await teardown({db, browser})
})

describe('dashboard', () => {
    it('redirects when unlogged', async () => {
        const browserPage = await browser.createPage()
        const status = await browserPage.open(`http://localhost:3000/dashboard`)
        expect(status).toBe('success')

        const text = await browserPage.property('content')
        const $ = cheerio.load(text)
        const page = $('#__next')
        const pageHtml = page.html()
        const html = pretty(pageHtml)
        expect(html).toMatchSnapshot()
    })

    it('displays when logged', async () => {
        const browserPage = await browser.createPage()

        await generateAndLogUser(browserPage)

        const status = await browserPage.open(`http://localhost:3000/dashboard`)
        expect(status).toBe('success')
        const text = await browserPage.property('content')
        const $ = cheerio.load(text)
        const page = $('#__next')
        const pageHtml = page.html()
        const html = pretty(pageHtml)
        expect(html).toMatchSnapshot()
    })
})
