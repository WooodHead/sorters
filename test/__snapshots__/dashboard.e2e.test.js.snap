// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard displays when logged 1`] = `
"<div data-reactroot=\\"\\">
  <div>
    <!-- react-empty: 3 -->
    <div style=\\"padding-top:50px;padding-bottom:20px;\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\">
        <div class=\\"container\\">
          <div class=\\"navbar-header\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\"><span class=\\"sr-only\\">Toggle navigation</span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span></button>
            <a class=\\"navbar-brand\\"
              href=\\"/\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\">
            <ul class=\\"nav navbar-nav navbar-right\\">
              <li class=\\"false\\">
                <a href=\\"/\\">Home</a>
              </li>
              <li class=\\"false\\">
                <a href=\\"/about\\">About</a>
              </li>
              <li class=\\"false\\">
                <a href=\\"/news\\">News</a>
              </li>
              <li class=\\"false\\">
                <a href=\\"/users\\">Sorters</a>
              </li>
              <li class=\\"active\\">
                <a href=\\"/account\\">Dashboard</a>
              </li>
              <li class=\\"dropdownfalse\\">
                <a href=\\"#\\" class=\\"dropdown-toggle\\" data-toggle=\\"dropdown\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"true\\">
                  Hello,
                  test
                  
                  <!-- /react-text --><span class=\\"caret\\"></span></a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div class=\\"container\\">
      <h1>Dashboard</h1>
      <p>
        Your profile can be found at
        <a href=\\"/u/test\\">/u/test</a>
        .
      </p>
      <div class=\\"alert alert-info\\">
        <p>Thank you for joining sorters club! This is your dashboard. As the platform develops, you will find more features here.</p>
      </div>
      <div class=\\"row\\">
        <div class=\\"col-md-4\\">
          <div class=\\"panel panel-default\\">
            <div class=\\"panel-heading\\">
              <h3 class=\\"panel-title\\">
                <a href=\\"/account/profile\\">&#x1F464; Profile</a>
              </h3>
            </div>
            <div class=\\"panel-body\\">
              <p>Edit your public profile.</p>
              <a class=\\"btn btn-default\\" href=\\"/account/profile\\" role=\\"button\\">Manage profile</a>
            </div>
          </div>
        </div>
        <div class=\\"col-md-4\\">
          <div class=\\"panel panel-default\\">
            <div class=\\"panel-heading\\">
              <h3 class=\\"panel-title\\">
                <a href=\\"/account/goals\\">&#x25CE; Goals</a>
              </h3>
            </div>
            <div class=\\"panel-body\\">
              <p>Share your goals and leave updates.</p>
              <a class=\\"btn btn-default\\" href=\\"/account/goals\\" role=\\"button\\">Manage goals</a>
            </div>
          </div>
        </div>
        <div class=\\"col-md-4\\">
          <div class=\\"panel panel-default\\">
            <div class=\\"panel-heading\\">
              <h3 class=\\"panel-title\\">
                <a href=\\"/account/journal\\">&#x270E; Journal</a>
              </h3>
            </div>
            <div class=\\"panel-body\\">
              <p>Write updates about your goals.</p>
              <a class=\\"btn btn-default\\" href=\\"/account/journal\\" role=\\"button\\">Manage Journal</a>
            </div>
          </div>
        </div>
      </div>
      <div class=\\"row\\">
        <div class=\\"col-md-4\\">
          <div class=\\"panel panel-default\\">
            <div class=\\"panel-heading\\">
              <h3 class=\\"panel-title\\">
                <a href=\\"/account/topics\\"><span>&#x1F4A1; Topics <!-- /react-text --><span class=\\"badge pull-right\\">new</span></span>
                </a>
              </h3>
            </div>
            <div class=\\"panel-body\\">
              <p>Share what topics you are interested in.</p>
              <a class=\\"btn btn-default\\" href=\\"/account/topics\\" role=\\"button\\">Manage Topics</a>
            </div>
          </div>
        </div>
        <div class=\\"col-md-4\\">
          <div class=\\"panel panel-default\\">
            <div class=\\"panel-heading\\">
              <h3 class=\\"panel-title\\">
                <a href=\\"/account/essays\\"><span>&#x270E; Essays <!-- /react-text --><span class=\\"badge pull-right\\">new</span></span>
                </a>
              </h3>
            </div>
            <div class=\\"panel-body\\">
              <p>Write essays about the topics you are interested in or about the books you read.</p>
              <a class=\\"btn btn-default\\" href=\\"/account/essays\\" role=\\"button\\">Manage Essays</a>
            </div>
          </div>
        </div>
        <div class=\\"col-md-4\\">
          <div class=\\"panel panel-default\\">
            <div class=\\"panel-heading\\">
              <h3 class=\\"panel-title\\">
                <a href=\\"/account/speeches\\"><span>&#x1F444; Speeches <!-- /react-text --><span class=\\"badge pull-right\\">new</span></span>
                </a>
              </h3>
            </div>
            <div class=\\"panel-body\\">
              <p>Give speeches about the topics you are interested in or about the books you read.</p>
              <a class=\\"btn btn-default\\" href=\\"/account/speeches\\" role=\\"button\\">Manage Speeches</a>
            </div>
          </div>
        </div>
      </div>
      <div class=\\"row\\">
        <div class=\\"col-md-4\\">
          <div class=\\"panel panel-default\\">
            <div class=\\"panel-heading\\">
              <h3 class=\\"panel-title\\">
                <a href=\\"/account/conversations\\"><span>&#x1F5E9; Conversations <!-- /react-text --><span class=\\"badge pull-right\\">new</span></span>
                </a>
              </h3>
            </div>
            <div class=\\"panel-body\\">
              <p>Have conversations with other sorters about topics you are interested in.</p>
              <a class=\\"btn btn-default\\" href=\\"/account/conversations\\" role=\\"button\\">Start Conversations</a>
            </div>
          </div>
        </div>
        <div class=\\"col-md-4\\">
          <div class=\\"panel panel-default\\">
            <div class=\\"panel-heading\\">
              <h3 class=\\"panel-title\\">
                <a href=\\"/account/reads\\">&#x1F4D6; Reading list</a>
              </h3>
            </div>
            <div class=\\"panel-body\\">
              <p>Create a list of things you read and intend to read.</p>
              <a class=\\"btn btn-default\\" href=\\"/account/reads\\" role=\\"button\\">Manage Reading List</a>
            </div>
          </div>
        </div>
        <div class=\\"col-md-4\\">
          <div class=\\"panel panel-default\\">
            <div class=\\"panel-heading\\">
              <h3 class=\\"panel-title\\">
                <a href=\\"/account/news\\">&#x1F46B; Sorters</a>
              </h3>
            </div>
            <div class=\\"panel-body\\">
              <p>See what other sorters are doing.</p>
              <a class=\\"btn btn-default\\" href=\\"/account/news\\" role=\\"button\\">Sorters</a>
            </div>
          </div>
        </div>
      </div>
      <div class=\\"row\\">
        <div class=\\"col-md-4\\">
          <div class=\\"panel panel-default\\">
            <div class=\\"panel-heading\\">
              <h3 class=\\"panel-title\\">
                <a href=\\"https://discord.gg/6Q8v9Sm\\">&#x1F5E9; Chat</a>
              </h3>
            </div>
            <div class=\\"panel-body\\">
              <p>Chat with other sorters (a Discord channel).</p>
              <a class=\\"btn btn-default\\" href=\\"https://discord.gg/6Q8v9Sm\\" target=\\"_blank\\" role=\\"button\\">Chat</a>
            </div>
          </div>
        </div>
        <div class=\\"col-md-4\\">
          <div class=\\"panel panel-default\\">
            <div class=\\"panel-heading\\">
              <h3 class=\\"panel-title\\">
                <a href=\\"/account/account\\">&#x2699; Account</a>
              </h3>
            </div>
            <div class=\\"panel-body\\">
              <p>Set your username, password etc.</p>
              <a class=\\"btn btn-default\\" href=\\"/account/account\\" role=\\"button\\">Manage Account</a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer>
      <div class=\\"container\\">
        <hr>
        <p>
          Brought to you with &#x2764; by
          <a href=\\"/about\\">Nick Redmark</a>
          . Support this platform on
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\">Patreon</a>
          .
        </p>
      </div>
    </footer>
  </div>
</div>"
`;

exports[`dashboard redirects when unlogged 1`] = `
"<div data-reactroot=\\"\\">
  <div>
    <!-- react-empty: 3 -->
    <div style=\\"padding-top:50px;padding-bottom:20px;\\">
      <nav class=\\"navbar navbar-inverse navbar-fixed-top\\">
        <div class=\\"container\\">
          <div class=\\"navbar-header\\">
            <button type=\\"button\\" class=\\"navbar-toggle collapsed\\" data-toggle=\\"collapse\\" data-target=\\"#navbar\\" aria-expanded=\\"false\\" aria-controls=\\"navbar\\"><span class=\\"sr-only\\">Toggle navigation</span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span><span class=\\"icon-bar\\"></span></button>
            <a class=\\"navbar-brand\\"
              href=\\"/\\">Sorters Club</a>
          </div>
          <div id=\\"navbar\\" class=\\"collapse navbar-collapse\\">
            <ul class=\\"nav navbar-nav navbar-right\\">
              <li class=\\"false\\">
                <a href=\\"/\\">Home</a>
              </li>
              <li class=\\"false\\">
                <a href=\\"/about\\">About</a>
              </li>
              <li class=\\"false\\">
                <a href=\\"/news\\">News</a>
              </li>
              <li class=\\"false\\">
                <a href=\\"/users\\">Sorters</a>
              </li>
              <li class=\\"active\\">
                <a href=\\"/account\\">Dashboard</a>
              </li>
              <li class=\\"false\\">
                <a href=\\"/login\\">Log in</a>
              </li>
              <li class=\\"false\\">
                <a href=\\"/register\\">Register</a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
    <div class=\\"container\\">
      <p>You need to log in to see this page.</p>
    </div>
    <footer>
      <div class=\\"container\\">
        <hr>
        <p>
          Brought to you with &#x2764; by
          <a href=\\"/about\\">Nick Redmark</a>
          . Support this platform on
          <a href=\\"https://www.patreon.com/nickredmark\\" target=\\"_blank\\">Patreon</a>
          .
        </p>
      </div>
    </footer>
  </div>
</div>"
`;
